server/images.go

```diff
@@ -708,6 +714,24 @@ func PullModel(name string, regOpts *RegistryOptions, fn func(api.ProgressRespon
 		return fmt.Errorf("pull model manifest: %s", err)
 	}
 
+	fn(api.ProgressResponse{Status: "writing manifest"})
```

Comment:
creating this before downloading the layers means we know which layers to delete in the case of a partial download

Reply:
Doesn't this cause the model to show up in `list` even if it's complete? Is that the right behaviour?

Reply:
It does show up in list when partially downloaded with this change. I'd say this behavior is ok, at least I know the partial files are on my filesystem in that case. It will just pull the rest when I run it.

Reply:
This feels... weird. I don't think most people would want to see that unless your download failed and you somehow forgot and wanted to restart it? I would think we would hide it unless you did something like `ollama ls -a`.

This did make me think of the `docker prune` command though. It would probably be good to have a command which cleans up the partial files and any dangling blobs which don't have a manifest associated with them. That can happen if there's a new version tag that you pull w/ different SHAs.